src/dense.jl

```diff
@@ -1,193 +1,209 @@
 # A higher level stepper, defined as a wrapper around another stepper.
 
-#TODO: how about having an DenseStepper <: AbstractWrapper <: AbstractStepper?
-immutable DenseStepper <: AbstractStepper
-    solver::Solver
+"""
+
+Dense output options:
+
+- tspan    ::Vector{T}  output times
+- points   ::Symbol which points are returned: `:specified` only the
+  ones in tspan or `:all` which includes also the step-points of the solver.
+- stopevent  Stop integration at a zero of this function
+- roottol    TODO
+
+"""
+
+immutable DenseOptions{T<:Number} <: Options{T}
+    tout::Vector{T}
+    # points   ::Symbol
+    # stopevent::S
+    # roottol  ::T
 end
 
-solve(ode::ExplicitODE,
-      stepper::DenseStepper,
-      options::Options) = Solver(ode,stepper,options)
+@compat function (::Type{DenseOptions{T}}){T}(;
+                                              tstop        = T(Inf),
+                                              tout::Vector = T[tstop],
+                                              # points::Symbol= :all,
+                                              # stopevent::S  = (t,y)->false,
+                                              # roottol       = eps(T)^T(1//3),
+                                              kargs...)
+    DenseOptions{T}(tout)
+end
 
-dense(sol::Solver) = solve(sol.ode, DenseStepper(sol), sol.options)
 
 """
 
-The state of the dense stepper
+A stepper specialized in dense output.  It wraps around another
+`Solver` and stores the subsequent steps generated by `Solver` and
+interpolates the results on request (currently this means at the
+output times stored in `options.tout`).
 
-- s0, s1: Previous steps, used to produce interpolated output
-- solver_state: The state of the associated solver
-- ytmp: work array
+"""
+immutable DenseStepper{S<:Solver,O<:DenseOptions} <: AbstractStepper
+    solver::S
+    options::O
+end
+
+function solve{T,S<:DenseStepper}(ode::ExplicitODE{T},
+                                  ::Type{S};
+                                  method = RKStepperAdaptive{:rk45},
```

Comment:
Hmm, maybe it would be clearer in general if the function `solve` was generically defined by calling the constructor of the `Stepper` and the stepper just defines an ordinary constructor?